### PR TITLE
[Doc] Add onboarding guide

### PR DESCRIPTION
**Pull request by joshuarrrr**
### Description

and link onboarding guide from README, CONTRIBUTING

### Issues Resolved

fixes https://github.com/opensearch-project/.github/issues/131

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ONBOARDING.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+- **Read (3 min) [the project homepage](https://opensearch.org/)** to understand to get a sense of what the OpenSearch software does and how it’s used
+- **Read (4 min) “[About OpenSearch](https://opensearch.org/about.html)”** to understand the primary project components and development principles
+- Optional - Read founding documents to understand the project history
+  - Read (8 min) “[Introducing OpenSearch](https://aws.amazon.com/blogs/opensource/introducing-opensearch/)” from 2021-04-12
+  - Read (6 min) “[Stepping up for a truly open source Elasticsearch](https://aws.amazon.com/blogs/opensource/stepping-up-for-a-truly-open-source-elasticsearch/)” from 2021-01-21
+
+### How we work
+
+We model our way of working on “[the Apache Way](https://apache.org/theapacheway)” (optional - read 4 min). Specifically, we’ve borrowed and adapted:
```
Comment (joshuarrrr) [inline ONBOARDING.md:38]: I would have preferred to just link to this as a reading, but it's pretty apache-specific and didn't capture the ways we've adapted for our project. But I could imagine this info living elsewhere. It's worth noting that this section is not original work, but I believe I've adapted and referenced it appropriately.

File: ONBOARDING.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+- [Frequently Asked Questions](https://opensearch.org/faq)
+- [User documentation](https://opensearch.org/docs/latest/)
+- Developer documentation
+  - Start with `DEVELOPER_GUIDE.md` in the repository you’ll be working in (see [OpenSearch](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md) or [OpenSearch Dashboards](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md) guides)
+  - Most repositories have many other documentation files. To find them, search for Markdown files (`*.md` - [example](https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch-Dashboards+language%3AMarkdown&type=code&l=Markdown)), either on GitHub or in your editor.
+
+## Set up accounts and access
+
+### GitHub account
```
Comment (joshuarrrr) [inline ONBOARDING.md:70]: Ideally we could just point to a single GitHub doc. But the GitHub getting started doc has a lot of extra content and noise that distract from the task. So I decided to go with a stripped down list here.

File: ONBOARDING.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+- [Learn key concepts](#learn-key-concepts)
+  - [About the OpenSearch project](#about-the-opensearch-project)
+  - [How we work](#how-we-work)
+  - [Contributor expectations](#contributor-expectations)
+  - [Resources and links](#resources-and-links)
+- [Set up accounts and access](#set-up-accounts-and-access)
+  - [GitHub account](#github-account)
+  - [OpenSearch forum account](#opensearch-forum-account)
+
+# Getting started as an OpenSearch contributor
```
Comment (dblock) [inline ONBOARDING.md:13]: Nit: can we Capitalize Titles Like in Other Markdown Files?

**Review by dblock — COMMENTED** [review]
I like this very much. 

@CEHENKLE Did we have another one of these somewhere? 

@joshuarrrr Do we link this from https://github.com/opensearch-project/.github/blob/main/profile/README.md?

File: ONBOARDING.md (hunk 3 above)
Comment (joshuarrrr) [reply ONBOARDING.md:13]: lol, sure - I'll also double check the style guide for other things I may have missed. Sentence case titles are a strong personal preference, but consistency is better 😉

Comment (AMoo-Miki) [reply ONBOARDING.md:13]: At least the CONTRIBUTING uses cap-titles.

File: ONBOARDING.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+## Learn key concepts
+
+### About the OpenSearch project
+
+- **Read (3 min) [the project homepage](https://opensearch.org/)** to understand to get a sense of what the OpenSearch software does and how it’s used
+- **Read (4 min) “[About OpenSearch](https://opensearch.org/about.html)”** to understand the primary project components and development principles
+- Optional - Read founding documents to understand the project history
+  - Read (8 min) “[Introducing OpenSearch](https://aws.amazon.com/blogs/opensource/introducing-opensearch/)” from 2021-04-12
+  - Read (6 min) “[Stepping up for a truly open source Elasticsearch](https://aws.amazon.com/blogs/opensource/stepping-up-for-a-truly-open-source-elasticsearch/)” from 2021-01-21
```
Comment (AMoo-Miki) [inline ONBOARDING.md:34]: I know it is the title of that article but as an adjective for "Elasticsearch", the term should be hyphenated as "open-source"; I don't think we should change it here unless they fix the origin.

File: ONBOARDING.md [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+
+Contributors are subject to a code of conduct, and it’s essential that all contributions are original work.
+
+- **Read (6 min)** [the Amazon Open Source Code of Conduct](https://aws.github.io/code-of-conduct)
+- **Read (6 min)** [the OpenSearch CONTRIBUTING guide](https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md)
+  - Please make sure you understand the implications of the [Developer Certificate of Origin](https://github.com/opensearch-project/.github/blob/main/CONTRIBUTING.md#developer-certificate-of-origin)
+
+### Resources and links
+
+There are more than 90 [GitHub repositories](https://github.com/opensearch-project/OpenSearch) that are part of the OpenSearch project. Explore the list to get a sense of the project breadth and scope.
```
Comment (AMoo-Miki) [inline ONBOARDING.md:56]: Should this be more than 100?

File: ONBOARDING.md (hunk 2 above)
Comment (AMoo-Miki) [reply ONBOARDING.md:70]: maybe put a link to that one to say if you need more information, see a complete version there.

File: ONBOARDING.md [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  - Most repositories have many other documentation files. To find them, search for Markdown files (`*.md` - [example](https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch-Dashboards+language%3AMarkdown&type=code&l=Markdown)), either on GitHub or in your editor.
+
+## Set up accounts and access
+
+### GitHub account
+
+To create issues, leave comments, or submit pull requests, you’ll need a GitHub account. Already have a GitHub account? Great! If not, it’s quick to get started
+
+1. **Sign up for an account** by navigating to https://github.com/ and following the prompts ([source](https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account#1-creating-an-account)).
+2. **[Verify your email address](https://docs.github.com/en/get-started/signing-up-for-github/verifying-your-email-address)**
```
Comment (AMoo-Miki) [inline ONBOARDING.md:75]: If the email is not made public, they cannot sign DCO. Evidenced by the number people who have missed this, IMO, we should point it out explicitly.

File: ONBOARDING.md [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+3. Recommended - [**Set up GitHub SSH keys**](https://docs.github.com/en/authentication/connecting-to-github-with-ssh/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent)
+4. Recommended - **[Set up 2-factor authentication](https://docs.github.com/en/authentication/securing-your-account-with-two-factor-authentication-2fa/about-two-factor-authentication)**
+5. Optional - If you’re new to using GitHub or git, we recommend reviewing “[Using GitHub's tools and processes](https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account#part-2-using-githubs-tools-and-processes)”
+
+### OpenSearch forum account
+
+There are multiple ways to [connect with the OpenSearch community](https://opensearch.org/connect.html), but [the forums](https://forum.opensearch.org/) are the best place to start
```
Comment (AMoo-Miki) [inline ONBOARDING.md:82]: They are called "forum" on top and "forums" here; which one is it?

File: ONBOARDING.md (hunk 3 above)
Comment (joshuarrrr) [reply ONBOARDING.md:13]: I'll defer to maintainers of this repo over whether we choose title casing to match other markdown files here, or whether it's better for new documents to follow the project style guide: https://github.com/opensearch-project/documentation-website/blob/main/STYLE_GUIDE.md#punctuation-and-capitalization

> Use sentence case for titles, headings, and table headers. Titles of standalone documents may use title case.

File: ONBOARDING.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+  - [GitHub account](#github-account)
+  - [OpenSearch forum account](#opensearch-forum-account)
+
+# Getting started as an OpenSearch contributor
+
+## Introduction
+
+### Who this is for
+
+Any developer who’s interested in contributing code (features and fixes) to one of the OpenSearch open source project repositories for the first time.
```
[the remark this answers is not in the document]
Comment (joshuarrrr) [reply ONBOARDING.md:19]: Huh, interesting. Seems like a bug/quirk in Quip's markdown export. Will replace.

File: ONBOARDING.md (hunk 4 above)
Comment (joshuarrrr) [reply ONBOARDING.md:34]: yeah, agreed, on both points. My intention here is to match the actual document titles of the source.

File: ONBOARDING.md (hunk 5 above)
Comment (joshuarrrr) [reply ONBOARDING.md:56]: This is only counting public repos (it excludes archived repos), which is currently 97. But the link is wrong. I'll update to https://github.com/orgs/opensearch-project/repositories?q=&type=public&language=&sort=

File: ONBOARDING.md [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+  - Start with `DEVELOPER_GUIDE.md` in the repository you’ll be working in (see [OpenSearch](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md) or [OpenSearch Dashboards](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/DEVELOPER_GUIDE.md) guides)
+  - Most repositories have many other documentation files. To find them, search for Markdown files (`*.md` - [example](https://github.com/search?q=repo%3Aopensearch-project%2FOpenSearch-Dashboards+language%3AMarkdown&type=code&l=Markdown)), either on GitHub or in your editor.
+
+## Set up accounts and access
+
+### GitHub account
+
+To create issues, leave comments, or submit pull requests, you’ll need a GitHub account. Already have a GitHub account? Great! If not, it’s quick to get started
+
+1. **Sign up for an account** by navigating to https://github.com/ and following the prompts ([source](https://docs.github.com/en/get-started/onboarding/getting-started-with-your-github-account#1-creating-an-account)).
```
[the remark this answers is not in the document]
Comment (joshuarrrr) [reply ONBOARDING.md:74]: Sure, will add.

File: ONBOARDING.md (hunk 6 above)
Comment (joshuarrrr) [reply ONBOARDING.md:75]: Is it really a requirement for contribution? Any developer who would prefer not to disclose their email publicly is de facto barred from contributing? If so, I definitely agree we need to highlight that (not just here, but in the DCO section of CONTRIBUTING)

File: ONBOARDING.md (hunk 7 above)
Comment (joshuarrrr) [reply ONBOARDING.md:82]: We're not consistent and use both interchangeably in this repo: https://github.com/search?q=repo%3Aopensearch-project%2F.github%20forum&type=code
Same thing in the style guide: https://github.com/opensearch-project/documentation-website/blob/main/STYLE_GUIDE.md#brand-personality-traits

If it's up to me, I prefer the singular and will update accordingly.

**Comment by joshuarrrr** [comment]
> @joshuarrrr Do we link this from https://github.com/opensearch-project/.github/blob/main/profile/README.md?

Added link in most recent commit.

**Review by peternied — DISMISSED** [review]
Thanks for working on this @joshuarrrr.  Looking at the unresolved threads most of these seems subtle, if there are strong opinions, lets follow up up with additional pull requests.

File: ONBOARDING.md (hunk 3 above)
Comment (peternied) [reply ONBOARDING.md:13]: I'd be happy to review a PR for a markdown linter if we'd like to codify these styles

**Review by peternied — APPROVED** [review]

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -94,7 +96,7 @@ Signed-off-by: Jane Smith <jane.smith@email.com>
 
 You may type this line on your own when writing your commit messages. However, if your `user.name` and `user.email` are set in your `git config`, you can use `-s` or `––signoff` to add the `Signed-off-by` line to the end of the commit message automatically.
```
Comment (ananzh) [inline CONTRIBUTING.md:97]: should be `--signoff` not `––signoff` right?

File: ONBOARDING.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+
+### How we work
+
+We model our way of working on “[the Apache Way](https://apache.org/theapacheway)” (optional - read 4 min). Specifically, we've borrowed and adapted:
+
+- *Community Over Code:* A healthy community is a higher priority than good code. Strong communities can always rectify problems with their code, whereas an unhealthy community will likely struggle to maintain a codebase in a sustainable manner.
+- *Earned Authority:* All individuals are given the opportunity to participate, but their influence is based on publicly earned merit – what they contribute to the community. Merit lies with the individual, does not expire, is not influenced by employment status or employer, and is non-transferable (merit earned in one project cannot be applied to another).
+- *Open Communications:* All communications related to code and decision-making should be publicly accessible to ensure asynchronous collaboration, as necessitated by a globally-distributed community.
+- *Community of Peers:* Individuals participate, not organizations. Roles are equal irrespective of title, votes hold equal weight, and contributions are made on a volunteer basis (even if paid to work on code).
```
Comment (ananzh) [inline ONBOARDING.md:43]: `a voluntary basis`?

File: ONBOARDING.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+- *Progress over Perfection:* Struggling with a test failure? Can't quite figure out why that data structure isn't doing what you expect? Feeling defeated by the complexities of that massively distributed compute engine? The community idea exists to help each other through difficult problems. So fail early, commit often, ask for help, and don't be afraid.
+
+### Contributor expectations
+
+Contributors are subject to a code of conduct, and it's essential that all contributions are original work.
+
+- **Read (6 min) [the Amazon Open Source Code of Conduct](https://aws.github.io/code-of-conduct)**
```
Comment (mch2) [inline ONBOARDING.md:50]: I think we should link the [OpenSearch code of conduct](https://opensearch.org/codeofconduct.html).

**Comment by dblock** [comment]
@joshuarrrr finish this?

**Comment by AMoo-Miki** [comment]
How would you feel about merging this and I will follow it up with the 2 changes asked for?

**Comment by dblock** [comment]
Let's close this one - @AMoo-Miki reopen a new one on top?

**Pull request closed without merging**